File: backend/src/models/Review.js
import mongoose from 'mongoose';

const reviewSchema = new mongoose.Schema({
  productId: { type: mongoose.Schema.Types.ObjectId, ref: 'Product', required: true, index: true },
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  userName: { type: String, required: true },
  rating: { type: Number, required: true, min: 1, max: 5 },
  title: { type: String, default: '' },
  comment: { type: String, required: true },
  isVerifiedPurchase: { type: Boolean, default: true },
  isApproved: { type: Boolean, default: true, index: true }
}, {
  timestamps: true
});

export const Review = mongoose.model('Review', reviewSchema);

File: app/src/main/java/com/housieshopping/app/domain/model/Review.kt
package com.housieshopping.app.domain.model

data class Review(
    val id: String,
    val productId: String,
    val userName: String,
    val userAvatarUrl: String? = null,
    val rating: Float,
    val title: String,
    val comment: String,
    val createdAt: String,
    val isVerifiedPurchase: Boolean = true,
    val imageUrls: List<String> = emptyList()
)
